VK/VK/DataProvider/PromiseProvider.swift

```diff
@@ -0,0 +1,41 @@
+// PromiseProvider.swift
+// Copyright © RoadMap. All rights reserved.
+
+import Foundation
+import PromiseKit
+import RealmSwift
+
+/// Выбор получения данных друзей
+final class PromiseProvider {
+    // MARK: Private Properties
+
+    private let networkService = NetworkService()
+    private let realmService = RealmService()
+    private let promiseService = PromiseService()
+    private var friendsToken: NotificationToken?
+
+    // MARK: - Public Methods
+
+    func observeFriends(handler: @escaping ([Friend]) -> ()) {
+        let results = realmService.fetch(Friend.self)
+        guard let results else { return }
+        friendsToken = results.observe { change in
+            switch change {
+            case .update, .initial:
+                handler(Array(results))
+            default:
+                break
+            }
+        }
+    }
+
+    func fetchFriends() {
+        firstly {
+            promiseService.fetchFriends()
+        }.done { [weak self] friends in
```

Comment:
```suggestion
        }.done {  friends in
```
https://github.com/mxcl/PromiseKit/blob/v6/Documentation/FAQ.md#do-i-need-to-worry-about-retain-cycles
в доке пишут, что  weak не требуется, но всегда можно пройтись и посмотреть лики самому